### PR TITLE
Add RDMA node discovery feature selector label

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: images/k8s-rdma-shared-dev-plugin-ds.yaml [hunk 1]
```diff
@@ -18,6 +18,9 @@ spec:
         name: rdma-shared-dp-ds
     spec:
       hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (adrianchiris) [inline images/k8s-rdma-shared-dev-plugin-ds.yaml:22]: on my setup i got: kubernetes.io/os=linux and not beta.kubernetes.io/os

File: images/k8s-rdma-shared-dev-plugin-ds-pre-1.16.yaml [hunk 2]
```diff
@@ -15,6 +15,9 @@ spec:
         name: rdma-shared-dp-ds
     spec:
       hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/os: linux
+        feature.node.kubernetes.io/custom-rdma.capable: "true"
```
Comment (adrianchiris) [inline images/k8s-rdma-shared-dev-plugin-ds-pre-1.16.yaml:20]: > feature.node.kubernetes.io/custom-rdma.capable: "true"
 this depends on NFD v0.6.0 (or manually marking the node with this label which i dont think we want)

i am not too sure we want to add this dependency here , but rather just add a section in README about how the nodes should be selected. (with a nodeSelector snippet)

File: README.md [hunk 3]
```diff
@@ -73,3 +73,21 @@ EOF
 ```
 kubectl create -f <rdma-app.yaml>
 ```
+
+# RDMA Shared Device Plugin with NFD (Node Feature Discovery)
+[NFD](https://github.com/kubernetes-sigs/node-feature-discovery) can be use to discovery the node capabilities, and expose them as labels to the node
+
+1. Deploy NDF
```
Comment (adrianchiris) [inline README.md:80]: NFD

File: README.md [hunk 4]
```diff
@@ -73,3 +73,21 @@ EOF
...
+
+```
+# export NFD_VERSION=v0.6.0
```
Comment (adrianchiris) [inline README.md:83]: its actually version v0.6.0 or newer. perhaps add a note

File: README.md [hunk 5]
```diff
@@ -73,3 +73,21 @@ EOF
 ```
 kubectl create -f <rdma-app.yaml>
 ```
+
+# RDMA Shared Device Plugin with NFD (Node Feature Discovery)
+[NFD](https://github.com/kubernetes-sigs/node-feature-discovery) can be use to discovery the node capabilities, and expose them as labels to the node
```
Comment (adrianchiris) [inline README.md:78]: it should be either: "can be used for discovery" or "can be used to discover" pick one :)

Comment (adrianchiris) [inline README.md:78]: as labels to the node -> as node labels

File: README.md [hunk 6]
```diff
@@ -73,3 +73,21 @@ EOF
...
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/$NFD_VERSION/nfd-master.yaml.template
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/$NFD_VERSION/nfd-worker-daemonset.yaml.template
+```
+
+2. Check the new labels added to the node
+```
+# kubectl get nodes --show-labels
+```
+
+The device plugin can work on nodes with label `feature.node.kubernetes.io/custom-rdma.capable=true`, which indicates that the node is RDMA capable.
```
Comment (adrianchiris) [inline README.md:93]: small rephrase suggestion: can work on -> is intended to work on

**Comment by Mmduh-483** [comment]
@adrianchiris  Updated the PR

File: README.md [hunk 7]
```diff
@@ -73,3 +73,21 @@ EOF
...
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/$NFD_VERSION/nfd-master.yaml.template
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/$NFD_VERSION/nfd-worker-daemonset.yaml.template
+```
+
+2. Check the new labels added to the node
+```
+# kubectl get nodes --show-labels
+```
+
+The device plugin is intended to work on nodes with label `feature.node.kubernetes.io/custom-rdma.capable=true`, which indicates that the node is RDMA capable.
```
Comment (moshe010) [inline README.md:93]: you don't want capable node you want rdam enabled which mean the RDMA modules are loaded

Comment (Mmduh-483) [reply README.md]: changed

File: README.md [hunk 8]
```diff
@@ -73,3 +73,21 @@ EOF
 ```
 kubectl create -f <rdma-app.yaml>
 ```
+
+# RDMA Shared Device Plugin with NFD (Node Feature Discovery)
```
Comment (adrianchiris) [inline README.md:77]: Repharase:

# RDMA shared device plugin deployment with node labels

RDMA shared device plugin should be deployed on nodes that:
1. Have RDMA capable hardware
2. RDMA kernel stack is loaded

To allow proper node selection [Node Feature Discovery (NFD)](https://github.com/kubernetes-sigs/node-feature-discovery)  can be used to discover the node capabilities, and expose them as node labels.

File: README.md [hunk 9]
```diff
@@ -73,3 +73,21 @@ EOF
...
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/$NFD_VERSION/nfd-master.yaml.template
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/$NFD_VERSION/nfd-worker-daemonset.yaml.template
+```
+
+2. Check the new labels added to the node
+```
+# kubectl get nodes --show-labels
+```
+
+The device plugin is intended to work on nodes with label `feature.node.kubernetes.io/custom-rdma.available=true`, which indicates that the node is RDMA capable and RDMA modules are loaded.
```
Comment (adrianchiris) [inline README.md:93]: Suggested rephrase for the last section

RDMA device plugin can then be deployed on nodes with `feature.node.kubernetes.io/custom-rdma.available=true`, which indicates that the node is RDMA capable and RDMA modules are loaded.

**Comment by adrianchiris** [comment]
LGTM

**Pull request merged**